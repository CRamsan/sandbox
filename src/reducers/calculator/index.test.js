import reducer from '.'

const initialState = {
  displayText: 'x^2 + y^2',
}

describe('calculator', () => {
  it('should return the initial state', () => {
    const next = reducer(initialState, {})
    expect(next).toEqual(initialState)
  })

  it('CALCULATOR_WRITE_TEXT', () => {
    const next = reducer(initialState, { type: 'CALCULATOR_WRITE_TEXT', text: '123 * cos(pi)' })
    expect(next).toEqual({ ...initialState, displayText: 'x^2 + y^2123 * cos(pi)' })
    expect(next).toMatchSnapshot()
  })

  it('CALCULATOR_BACKSPACE', () => {
    const next = reducer(initialState, { type: 'CALCULATOR_BACKSPACE' })
    expect(next).toEqual({ ...initialState, displayText: 'x^2 + y^' })
    expect(next).toMatchSnapshot()
  })

  it('CALCULATOR_CLEAR_TEXT', () => {
    const next = reducer(initialState, { type: 'CALCULATOR_CLEAR_TEXT' })
    expect(next).toEqual({ ...initialState, displayText: '' })
    expect(next).toMatchSnapshot()
  })
})
