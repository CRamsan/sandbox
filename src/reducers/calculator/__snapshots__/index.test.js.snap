// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculator CALCULATOR_BACKSPACE 1`] = `
Object {
  "displayText": "x^2 + y^",
}
`;

exports[`calculator CALCULATOR_CLEAR_TEXT 1`] = `
Object {
  "displayText": "",
}
`;

exports[`calculator CALCULATOR_WRITE_TEXT 1`] = `
Object {
  "displayText": "x^2 + y^2123 * cos(pi)",
}
`;
